core/src/main/java/com/linecorp/armeria/server/http/auth/OAuth1aToken.java

```diff
@@ -0,0 +1,216 @@
+package com.linecorp.armeria.server.http.auth;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Sets;
+
+/**
+ * The bearer token of <a href="https://oauth.net/core/1.0a/#anchor12">OAuth 1.0a authentication</a>.
+ * */
+public final class OAuth1aToken {
+
+    private static final Joiner.MapJoiner mapJoiner = Joiner.on(", ").withKeyValueSeparator("=");
+
+    /**
+     * realm parameter. (optional)
+     * */
+    private static final String REALM = "realm";
+
+    /**
+     * oauth_consumer_key parameter.
+     * */
+    private static final String OAUTH_CONSUMER_KEY = "oauth_consumer_key";
+
+    /**
+     * oauth_token parameter.
+     * */
+    private static final String OAUTH_TOKEN = "oauth_token";
+
+    /**
+     * oauth_signature_method parameter.
+     * */
+    private static final String OAUTH_SIGNATURE_METHOD = "oauth_signature_method";
+
+    /**
+     * oauth_signature parameter.
+     * */
+    private static final String OAUTH_SIGNATURE = "oauth_signature";
+
+    /**
+     * oauth_timestamp parameter.
+     * */
+    private static final String OAUTH_TIMESTAMP = "oauth_timestamp";
+
+    /**
+     * oauth_nonce parameter.
+     * */
+    private static final String OAUTH_NONCE = "oauth_nonce";
+
+    /**
+     * version parameter. (optional)
+     * If not set, the default value is 1.0.
+     * */
+    private static final String OAUTH_VERSION = "version";
+
+    /**
+     * Set of required parameters.
+     * */
+    private static final Set<String> REQUIRED_PARAM_KEYS = Sets.newHashSet(OAUTH_CONSUMER_KEY, OAUTH_TOKEN,
+                                                                           OAUTH_SIGNATURE_METHOD,
+                                                                           OAUTH_SIGNATURE, OAUTH_TIMESTAMP,
+                                                                           OAUTH_NONCE);
+
+    /**
+     * Set of optional parameters.
+     * */
+    private static final Set<String> OPTIONAL_PARAM_KEYS = Sets.newHashSet(REALM, OAUTH_VERSION);
+
+    /**
+     * Set of defined parameters, regardless of it is required or not.
+     * */
+    private static final Set<String> DEFINED_PARAM_KEYS = Sets.union(REQUIRED_PARAM_KEYS, OPTIONAL_PARAM_KEYS);
+
+    /**
+     * Creates a new {@link OAuth1aToken} from given arguments.
+     * */
+    public static OAuth1aToken of(Map<String, String> params) {
+        return new OAuth1aToken(params);
+    }
+
+    /**
+     * Map of parameters.
+     * */
```

Comment:
This comment is redundant. The declaration already tells us it's a map of params.